src/Source.jl

```diff
@@ -0,0 +1,175 @@
+"""
+`SQLite.Source` implementes the `DataStreams` framework for interacting with SQLite databases
+"""
+type Source <: Data.Source # <: IO
+    schema::Data.Schema
+    stmt::Stmt
+    status::Cint
+end
+"""
+Independently constructs an `SQLite.Source` in `db` with the SQL statement `sql`.
+Will bind `values` to any parameters in `sql`.
+`rows` is used to indicate how many rows to return in the query result if known beforehand. `rows=0` (the default) will return all possible rows.
+`stricttypes=false` will remove strict column typing in the result set, making each column effectively `Vectot{Any}`
+"""
+function Source(db::DB,sql::AbstractString, values=[];rows::Int=0,stricttypes::Bool=true)
+    stmt = SQLite.Stmt(db,sql)
+    bind!(stmt, values)
+    status = SQLite.execute!(stmt)
+    cols = SQLite.sqlite3_column_count(stmt.handle)
+    header = Array(UTF8String,cols)
+    types = Array(DataType,cols)
+    for i = 1:cols
+        header[i] = bytestring(SQLite.sqlite3_column_name(stmt.handle,i-1))
+        # do better column type inference; query what the column was created for?
+        types[i] = stricttypes ? SQLite.juliatype(stmt.handle,i) : Any
+    end
+    # rows == -1 && count(*)?
```

Comment:
yeah, that's how I was leaning as well. I think it would probably be fine for `select * from table` type queries, but when things start getting messier with joins and what not, you're just asking for trouble